Libraries/LibWeb/HTML/HTMLDialogElement.cpp

```diff
@@ -478,4 +489,90 @@ void HTMLDialogElement::invoker_command_steps(DOM::Element& invoker, String& com
+    if (auto const* target_dialog = as_if<HTMLDialogElement>(*target); target_dialog
+        && target_dialog->has_attribute(AttributeNames::open)
+        && target_dialog->is_modal()
+        && !target_dialog->get_bounding_client_rect().to_type<double>().contains(event.client_x(), event.client_y()))
+        return {};
+
+    // 3. Let currentNode be target.
+    auto current_node = target;
+
+    // 4. While currentNode is not null:
+    while (current_node) {
+        // 1. If currentNode is a dialog element and currentNode has an open attribute, then return currentNode.
+        if (auto const* current_dialog = as_if<HTMLDialogElement>(*current_node); current_dialog
+            && current_dialog->has_attribute(AttributeNames::open))
+            return as<HTMLElement>(*current_node);
+
+        // 2. Set currentNode to currentNode's parent in the flat tree.
+        current_node = current_node->shadow_including_first_ancestor_of_type<HTMLElement>();
+    }
+
+    // 5. Return null.
+    return {};
+}
+
+// https://html.spec.whatwg.org/multipage/interactive-elements.html#light-dismiss-open-dialogs
+void HTMLDialogElement::light_dismiss_open_dialogs(UIEvents::PointerEvent const& event, const GC::Ptr<DOM::Node> target)
+{
+    // To light dismiss open dialogs, given a PointerEvent event:
+
+    // 1. Assert: event's isTrusted attribute is true.
+    VERIFY(event.is_trusted());
+
+    // 2. Let document be event's target's node document.
+    auto& document = target->document();
+
+    // 3. If document's open dialogs list is empty, then return.
+    if (document.open_dialogs_list().is_empty())
+        return;
+
+    // 4. Let ancestor be the result of running nearest clicked dialog given event.
+    auto const ancestor = nearest_clicked_dialog(event, target);
+
+    // 5. If event's type is "pointerdown", then set document's dialog pointerdown target to ancestor.
+    if (event.type() == UIEvents::EventNames::pointerdown)
+        document.set_dialog_pointerdown_target(ancestor);
+
+    // 6. If event's type is "pointerup", then:
+    if (event.type() == UIEvents::EventNames::pointerup) {
+        // 1. Let sameTarget be true if ancestor is document's dialog pointerdown target.
+        bool const same_target = ancestor == document.dialog_pointerdown_target();
+
+        // 2. Set document's dialog pointerdown target to null.
+        document.set_dialog_pointerdown_target({});
+
+        // 3. If sameTarget is false, then return.
+        if (!same_target)
+            return;
+
+        // 4. Let topmostDialog be the last element of document's open dialogs list.
+        auto const topmost_dialog = document.open_dialogs_list().last();
+
+        // 5. If ancestor is topmostDialog, then return.
+        if (ancestor == topmost_dialog)
+            return;
+
+        // FIXME: 6. If topmostDialog's computed closed-by state is not Any, then return.
```

Comment:
This isn't a FIXME. :^)

Reply:
I left it as a FIXME because I'm not actually using the "computed closed-by state" algorithm's steps, just a simplification of it. I should add a comment explaining that (or better, just implement the algorithm).

Reply:
Ah, yeah some explanation would be good!